### ISSUE TITLE
TraceInfo.SqlText adapted for Postgres

### ISSUE THREAD

**Issue by feeleen** [linq2db/linq2db#3878]
Hi, 
Is there any chance to get postgres-adapted ```SqlText``` in a ```TraceInfo```?

when I use OnTrace with postgres database

```
LinqToDB.Data.DataConnection.TraceSwitch.Level = TraceLevel.Verbose;
LinqToDB.Data.DataConnection.OnTrace = (TraceInfo trace) =>
            {
                if (trace.TraceInfoStep == TraceInfoStep.Error)
                {
                    Console.WriteLine(trace.SqlText);
                }
            };
```

I get sql server-related sql string, not the postgres

```
--  PostgreSQL.9.5 PostgreSQL
DECLARE @param Integer -- Int32
SET     @param = 1

INSERT INTO foo VALUES (@param);
```

and when parameter is int array (for example), array values not expanded and there's a strange type value -2147483639 :

```
DECLARE @param -2147483639 -- Object
SET     @param = System.Int32[]

INSERT INTO foo VALUES (param);
```

**Comment by feeleen** [comment]
maybe it could look like that code?

```sql
do
$$
--  PostgreSQL.9.5 PostgreSQL
declare 
	paramA integer = 1;
	paramB integer[] = array[2,1];
begin
	INSERT INTO foo VALUES (paramA, paramB);
end
$$
```

**Comment by MaceWindu** [comment]
This was already rised many times, e.g. #3833

We don't have plans to do it

**Comment by feeleen** [comment]
@MaceWindu 
Does it make sense to introduce some kind of ```ISqlTextVerboseBuilder```?
Then I'll plug in my implementation via configuration

**Comment by feeleen** [comment]
I just want to see execute-ready script in ```TraceInfo.SqlText```, to be able to rerun the exact same query with all the parameters in a separate database client query tool.

I suggested to introduce some pluggable formatter there, just to not do the same work twice (once for sql server which is already made there, and the second for my own implementation) and to cleanup this current useless sqlText (if db is not sql server). If i use postgres i'll write my own formatter and plug it in => profit.

Complex types not widely used, and I believe that they can be represented in json format with some helper functions. 

You are right, I can generate my own script now without limitations, so you can close this issue if it doesn't look reasonable.

**Comment by MaceWindu** [comment]
Yep, this is what I expected to hear. I will leave it open for now to collect more opinions wether we should customize this area.
